website/docs/r/iot_security_solution.html.markdown

```diff
@@ -0,0 +1,134 @@
+
+* `enabled` - (Optional) Is the Iot Security Solution enabled? Defaults to `true`.
+
+* `unmasked_ip_logging_enabled` - (Optional) Is unmasked IP logging enabled? Defaults to `false`.
+
+* `export` - (Optional) A list of data which is to exported to analytic workspace. Valid values include `RawEvents`.
+
+* `recommendation` - (Optional) A `recommendation` block as defined below.
+
+* `user_defined_resource` - (Optional) A `user_defined_resource` block as defined below.
+
+* `tags` - (Optional) A mapping of tags to assign to the resource.
+
+---
+
+A `recommendation` block supports the following:
+
+* `iot_acr_authentication_enabled` - (Optional) Could Service Principal Authentication be used with ACR repository? Defaults to `true`.
+
+* `iot_agent_send_unutilized_msg_enabled` - (Optional) Could Agent send underutilized messages? Defaults to `true`.
+
+* `iot_baseline_enabled` - (Optional) Is Security related system configuration issues identified? Defaults to `true`.
+
+* `iot_edge_hub_mem_optimize_enabled` - (Optional) Is IoT Edge Hub memory optimized? Defaults to `true`.
+
+* `iot_edge_logging_option_enabled` - (Optional) Is logging configured for IoT Edge module? Defaults to `true`.
+
+* `iot_inconsistent_module_settings_enabled` - (Optional) Does SecurityGroup has inconsistent module settings? Defaults to `true`.
+
+* `iot_install_agent_enabled` - (Optional) is Azure IoT Security agent installed? Defaults to `true`.
+
+* `iot_ip_filter_deny_all_enabled` - (Optional) Is Default IP filter policy denied? Defaults to `true`.
+
+* `iot_ip_filter_permissive_rule_enabled` - (Optional) Is IP filter rule source allowable IP range too large? Defaults to `true`.
+
+* `iot_open_ports_enabled` - (Optional) Is any ports open on the device? Defaults to `true`.
+
+* `iot_permissive_firewall_policy_enabled` - (Optional) Does firewall policy exist which allow necessary communication to/from the device? Defaults to `true`.
+
+* `iot_permissive_input_firewall_rules_enabled` - (Optional) Is only necessary addresses or ports are permitted in? Defaults to `true`.
+
+* `iot_permissive_output_firewall_rules_enabled` - (Optional) Is only necessary addresses or ports are permitted out? Defaults to `true`.
+
+* `iot_privileged_docker_options_enabled` - (Optional) Is high level permissions are needed for the module? Defaults to `true`.
+
+* `iot_shared_credentials_enabled` - (Optional) Is any credentials shared among devices? Defaults to `true`.
+
+* `iot_vulnerable_tls_cipher_suite_enabled` - (Optional) Does TLS cipher suite need to be updated? Defaults to `true`.
+
+---
+
+A `user_defined_resource` block supports the following:
+
+* `query` - (Required) Azure Resource Graph query which represents the security solution's user defined resources.
+
+* `subscription_ids` - (Required) A list of subscription Ids on which the user defined resources query should be executed..
```

Comment:
I think we should improve the ux here, i'm not sure what this is doing exactly? a list of iot hubs to apply this to via a graph query?

Reply:
here is the doc about `user_defined_resource`: https://docs.microsoft.com/en-us/azure/defender-for-iot/quickstart-configure-your-solution. Through this field, user could monitor the entire IoT solution in one dashboard, surfacing all of your IoT devices, IoT platforms and back-end resources in Azure. 

this is not for auto find iothub. In the portal, the ux is:
![tempsnip](https://user-images.githubusercontent.com/2786738/105442786-5a57e100-5ca5-11eb-8ff1-b617aaf60c0c.png)

user choose the sub or resourceGroup, the portal will convert into graph query and send to the backend. Should we do like the portal and dynamically build the graph?

Reply:
Maybe it would be best to rename these to query_for_resources &  query_subscription_ids?

Reply:
updated